gcc/rust/analysis/rust-hir-liveness.cc

```diff
@@ -65,33 +65,29 @@ Liveness::Analysis (HIR::Crate &crate)
 void
 Liveness::go (HIR::Crate &crate)
 {
+  CrateNum crateNum = crate.get_mappings ().get_crate_num ();
   while (!worklist.empty ())
     {
       HirId hirId = worklist.back ();
       worklist.pop_back ();
       scannedSymbols.emplace (hirId);
-      HIR::Item *item
-	= mappings->lookup_hir_item (crate.get_mappings ().get_crate_num (),
-				     hirId);
-      if (item == nullptr)
-	continue;
+      HIR::Item *item = mappings->lookup_hir_item (crateNum, hirId);
       liveSymbols.emplace (hirId);
-      item->accept_vis (*this);
+      if (item != nullptr)
+	{
+	  item->accept_vis (*this);
+	}
+      else
+	{ // the item maybe inside a trait impl
+	  HirId parent_impl_id = UNKNOWN_HIRID;
```

Comment:
This is something i hope we can fix soon. Where you look up an item but give it a visitor instead of trying to do lookup_{item,implitem... etc} so that it can be done in one go. It might be worth fixing as part of your work here. I will make an issue for it. see #365

Reply:
Okay, do you mean #366? I'll take a look.

Reply:
yeah I don't mean to fix that as part of this PR but it was something on my mind.